apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -62,6 +62,15 @@ internal SubscriberClientImpl(SubscriptionName subscriptionName, IEnumerable<Sub
         _modifyDeadlineSeconds = (int)((settings.AckDeadline ?? DefaultAckDeadline).TotalSeconds);
         var autoExtendInterval = TimeSpan.FromSeconds(_modifyDeadlineSeconds) - (settings.AckExtensionWindow ?? DefaultAckExtensionWindow);
         var autoExtendIntervalForExactlyOnceDelivery = settings.AckExtensionWindow ?? MinimumAckExtensionWindowForExactlyOnceDelivery;
```

Comment:
These two here have a different meaning, so their names definetely shouldn't be similar (and maybe other things shouldn't be similar). If these two values are being used for the same purpose elsewhere, that's certainly going to yield wrong results.

autoExtendInterval => is the delay before we attempt to extende the lease.
autoExtendIntervalForExactlyOnceDelivery => is the window we have for extending the lease.

Reply:
While I agree that names can be better, I just thought, I will add more context here:
autoExtendInterval -> It's the interval after which lease of unhandled messages is automatically extended for non-exactly once subscriptions
autoExtendIntervalForExactlyOnceDelivery -> Is the same, but for exactly-once delivery subscriptions. It is just that the interval calculation comes from AckExtensionWindow.

Reply:
No, it's not the same. `settings.AckExtensionWindow` is "How long we're meant to allow before the deadline".

Put it this way: if someone increases `settings.AckExtensionWindow`, then in non-EOD that means we extend the lease earlier; in EOD that means we extend the lease later. That's not right!

Reply:
Thanks Jon and Amanda. This is now addressed.